### PR TITLE
clockref improvements

### PR DESCRIPTION
**Pull request by daschuer**
This is a bandaid for:
https://bugs.launchpad.net/mixxx/+bug/1535746

If you use a mic and headphones only, it can happen that Mixxx picks the mic soundcard as clock reference which leads to noise. 
Now Mixxx picks only an input only soundcard if there is no output at all.

### CODE REVIEW

File: src/soundio/soundmanager.h [hunk 1]
```diff
@@ -113,6 +113,13 @@ class SoundManager : public QObject {
     void inputRegistered(AudioInput input, AudioDestination *dest);
 
   private:
+
+    struct DeviceMode {
```
Comment (rryan) [inline src/soundio/soundmanager.h]: Does it need to be in the header?

File: src/soundio/soundmanager.cpp [hunk 2]
```diff
@@ -416,22 +419,23 @@ Result SoundManager::setupDevices() {
...
-        QPair<bool, bool> mode(toOpen[device]);
-        bool isInput = mode.first;
-        bool isOutput = mode.second;
+    for (const auto& mode: toOpen) {
         ++devicesAttempted;
+        SoundDevice* device = mode.device;
         m_pErrorDevice = device;
         // If we have not yet set a clock source then we use the first
+        // output device
+        qDebug() << haveOutput << mode.isOutput;
```
Comment (rryan) [inline src/soundio/soundmanager.cpp]: Did you mean to include the debug line?

Comment (rryan) [inline src/soundio/soundmanager.cpp]: Did you mean to include the debug line?

**Comment by rryan** [comment]
LGTM other than my comments.

**Comment by daschuer** [comment]
Thank you for review

**Pull request merged**